### PR TITLE
KAFKA-15853: Move general configs out of KafkaConfig

### PR DESCRIPTION
**Pull request by OmniaGM**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by OmniaGM** [comment]
@nizhikov can you have a look please when you have time? There are only 2 PRs left for this work one to move `configDef` and then the final one will be to move `KafkaConfig` class

File: server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public static final String DELEGATION_TOKEN_MAX_LIFETIME_CONFIG = "delegation.token.max.lifetime.ms";
+    public static final long DELEGATION_TOKEN_MAX_LIFE_TIME_MS_DEFAULT = 7 * 24 * 60 * 60 * 1000L;
+    public static final String DELEGATION_TOKEN_MAX_LIFE_TIME_DOC = "The token has a maximum lifetime beyond which it cannot be renewed anymore. Default value 7 days.";
+
+    public static final String DELEGATION_TOKEN_EXPIRY_TIME_MS_CONFIG = "delegation.token.expiry.time.ms";
+    public static final long DELEGATION_TOKEN_EXPIRY_TIME_MS_DEFAULT = 24 * 60 * 60 * 1000L;
+    public static final String DELEGATION_TOKEN_EXPIRY_TIME_MS_DOC = "The token validity time in milliseconds before the token needs to be renewed. Default value 1 day.";
+
+    public static final String DELEGATION_TOKEN_EXPIRY_CHECK_INTERVAL_MS_CONFIG = "delegation.token.expiry.check.interval.ms";
+    public static final long DELEGATION_TOKEN_EXPIRY_CHECK_INTERVAL_MS_DEFAULT = 1 * 60 * 60 * 1000L;
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java]: `1 * 60 * 60 * 1000L;` -> `60 * 60 * 1000L;`

File: server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+package org.apache.kafka.server.config;
+
+public class DelegationTokenManagerConfigs {
+    /** ********* Delegation Token Configuration ****************/
+    public static final String DELEGATION_TOKEN_SECRET_KEY_CONFIG = "delegation.token.secret.key";
+    public static final String DELEGATION_TOKEN_SECRET_KEY_DOC = "Secret key to generate and verify delegation tokens. The same key must be configured across all the brokers. " +
+            " If using Kafka with KRaft, the key must also be set across all controllers. " +
+            " If the key is not set or set to empty string, brokers will disable the delegation token support.";
+
+    public static final String DELEGATION_TOKEN_SECRET_KEY_ALIAS_CONFIG = "delegation.token.master.key";
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java:27]: This is unrelated to this PR, but maybe we should add `@deprecated` for it.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 3]
```diff
@@ -1519,7 +1411,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
     require(!interBrokerUsesSasl || saslEnabledMechanisms(interBrokerListenerName).contains(saslMechanismInterBrokerProtocol),
       s"${KafkaSecurityConfigs.SASL_MECHANISM_INTER_BROKER_PROTOCOL_CONFIG} must be included in ${KafkaSecurityConfigs.SASL_ENABLED_MECHANISMS_CONFIG} when SASL is used for inter-broker communication")
     require(queuedMaxBytes <= 0 || queuedMaxBytes >= socketRequestMaxBytes,
-      s"${KafkaConfig.QueuedMaxBytesProp} must be larger or equal to ${SocketServerConfigs.SOCKET_RECEIVE_BUFFER_BYTES_CONFIG}")
+      s"${KafkaServerConfigs.QUEUED_MAX_BYTES_CONFIG} must be larger or equal to ${SocketServerConfigs.SOCKET_RECEIVE_BUFFER_BYTES_CONFIG}")
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: This is unrelated to this PR, but could you please change `SOCKET_RECEIVE_BUFFER_BYTES_CONFIG` to `SOCKET_REQUEST_MAX_BYTES_CONFIG`?

File: server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    public static final String DELETE_TOPIC_ENABLE_DOC = "Enables delete topic. Delete topic through the admin tool will have no effect if this config is turned off";
+
+    public static final String COMPRESSION_TYPE_CONFIG = ServerTopicConfigSynonyms.serverSynonym(TopicConfig.COMPRESSION_TYPE_CONFIG);
+    public static final String COMPRESSION_TYPE_DOC = "Specify the final compression type for a given topic. This configuration accepts the standard compression codecs " +
+            "('gzip', 'snappy', 'lz4', 'zstd'). It additionally accepts 'uncompressed' which is equivalent to no compression; and " +
+            "'producer' which means retain the original compression codec set by the producer.";
+
+    public static final String COMPRESSION_GZIP_LEVEL_CONFIG = ServerTopicConfigSynonyms.serverSynonym(TopicConfig.COMPRESSION_GZIP_LEVEL_CONFIG);
+    public static final String COMPRESSION_GZIP_LEVEL_DOC = "The compression level to use if " + COMPRESSION_TYPE_CONFIG + " is set to 'gzip'.";
+    public static final String COMPRESSION_LZ_4_LEVEL_CONFIG = ServerTopicConfigSynonyms.serverSynonym(TopicConfig.COMPRESSION_LZ4_LEVEL_CONFIG);
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java]: Could we align the naming? `COMPRESSION_LZ_4_LEVEL_CONFIG` -> `COMPRESSION_LZ4_LEVEL_CONFIG`

**Review by chia7712 — COMMENTED** [review]
@OmniaGM nice patch! happy to see `KafkaConfig` gets smaller :)

File: server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java (hunk 1 above)
Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java]: I think `1` was added originally to follow one of the config patterns like for configs like `DEFAULT_DELETE_RETENTION_MS` (24 * 60 * 60 * 1000L)  or `LOG_CLEANUP_INTERVAL_MS_DEFAULT`(5 * 60 * 1000). I removed it as there is no one pattern for representing unit of `1`.

File: server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java (hunk 2 above)
Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/DelegationTokenManagerConfigs.java:27]: Marked it as deprecated as the doc has been indicating it is deprecated for long time

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 3 above)
Comment (OmniaGM) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: Nice catch, I just translated this part and didn't think of rechecking it.

File: server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java (hunk 4 above)
Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java]: done, published a fix

**Comment by chia7712** [comment]
```
> Task :core:compileTestScala
[Error] /home/chia7712/project/kafka/core/src/test/scala/unit/kafka/server/LogDirFailureTest.scala:57:45: value ControlledShutdownEnableProp is not a member of object kafka.server.KafkaConfig
one error found
```
please fix the build error

File: server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.config;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.server.authorizer.Authorizer;
+public class KafkaServerConfigs {
```
Comment (mimaison) [inline server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java]: Do we need the `Kafka` prefix? Could we name this file `ServerConfigs` or `BrokerConfigs`?

Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java]: Not needed but we have already `KafkaServer` so I named it as `KafkaServerConfigs` to follow that.

Comment (mimaison) [reply server/src/main/java/org/apache/kafka/server/config/KafkaServerConfigs.java]: I think we can drop the prefix and have shorter names. KafkaProducer uses ProducerConfig.

**Comment by chia7712** [comment]
@OmniaGM It seems the conflicts are not fixed.

**Comment by OmniaGM** [comment]
> @OmniaGM It seems the conflicts are not fixed.

this is a new one :D will fix this now

**Comment by OmniaGM** [comment]
Fixed now!

**Review by mimaison — APPROVED** [review]
LGTM, thanks

**Pull request merged**